src/string.rs

```diff
@@ -0,0 +1,85 @@
+use core::slice;
+
+//Reusable type for chars, in xv6 they appear to be
+//signed chars by default
+#[allow(non_camel_case_types)]
+pub type c_char = i8;
+
+
+#[no_mangle]
+pub extern "C" fn strlen(s: *const c_char) -> isize {
+    unsafe {
+        let mut n = 0;
+        while *s.offset(n) != 0 {
+            n += 1;
+        }
+        n
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn strncmp(s: *const c_char, 
+                          t: *const c_char, 
+                          n: usize) 
+                          -> isize {
+
+    //This will give weird results if you pass in a negative number
+    let s_slice = unsafe {slice::from_raw_parts(s,n as usize)};
+    let t_slice = unsafe {slice::from_raw_parts(t,n as usize)};
```

Comment:
I'm still being a bit nit picky but you dont need `n as usize` here anymore since `n` is now a `usize`.